signxml/__init__.py

```diff
@@ -389,9 +395,10 @@ def sign(self, method=methods.enveloped, algorithm="rsa-sha256", key=None, passp
                 signature = long_to_bytes(r).rjust(32, b"\0") + long_to_bytes(s).rjust(32, b"\0")
 
             signature_value.text = ensure_str(b64encode(signature))
-
             key_info = SubElement(self.sig_root, ds_tag("KeyInfo"))
-            if cert_chain is None:
+            if key_name is not None:
+                SubElement(key_info, ds_tag("KeyName")).text = key_name
+            elif cert_chain is None:
```

Comment:
These conditions are not mutually exclusive. This has to be an `if`, not an `elif`, if I understand correctly.